ddtrace/internal/processor/trace.py

```diff
@@ -332,40 +332,41 @@ def on_span_finish(self, span):
                     break
 
 
-class PeerServiceProcessor(SpanProcessor):
+class PeerServiceProcessor(TraceProcessor):
     def __init__(self, peer_service_config):
         self._config = peer_service_config
         self._set_defaults_enabled = self._config.set_defaults_enabled
+        self._mapping = self._config.peer_service_mapping
 
-    def on_span_start(self, span):
-        """
-        We don't do anything on span start
-        """
-        pass
-
-    def on_span_finish(self, span):
-        if self._set_defaults_enabled:
-            self._set_span_tag(span)
-        self._remap_peer_service(span)  # Remap regardless of whether defaults are enabled
-
-    def _set_span_tag(self, span):
-        if span.get_tag(self._config.tag_name):  # If the tag already exists, assume it is user generated
-            span.set_tag_str(self._config.source_tag_name, self._config.tag_name)
+    def process_trace(self, trace):
+        if not trace:
             return
 
-        if span.get_tag(SPAN_KIND) not in self._config.enabled_span_kinds:
-            return
+        traces_to_process = []
```

Comment:
This should be called `spans_to_process` right? since this function takes a single trace/`List[Span]`